src/services/fdp-storage/fdp-storage.utils.ts

```diff
@@ -27,18 +28,37 @@ function extractDappIdFromRegex(url: string, regex: RegExp): string | null {
   return null
 }
 
-export function dappUrlToId(url: string, beeUrl: string): DappId {
-  assertBeeUrl(beeUrl)
+function extractDappIdFromWhitelistedDapps(url: string): string | undefined {
+  const { dappId } = whitelistedDapps.find(({ url: dappUrl }) => url.startsWith(dappUrl)) || {}
 
-  let dappId = extractDappIdFromRegex(url, constructBzzRegex(beeUrl))
+  return dappId
+}
 
-  if (dappId) {
-    return dappId
-  }
+function isDappIdForged(dappId: string, url: string): boolean {
+  let isWhitelistedDappId = false
+
+  const matchingUrl = whitelistedDapps.some(({ url: currentUrl, dappId: currentDappId }) => {
+    if (currentDappId === dappId) {
+      isWhitelistedDappId = true
+
+      return url.startsWith(currentUrl)
+    }
+
+    return false
+  })
+
+  return isWhitelistedDappId && !matchingUrl
```

Comment:
Maybe I'm missing smth but it should be enough.

```suggestion
  return !whitelistedDapps.some(({ url: currentUrl, dappId: currentDappId }) => {
      return currentDappId === dappId && url.startsWith(currentUrl)
  })
```

Reply:
With this change, if `dappId` doesn't match any of the `currentDappId`, the `some` function will return `false`. And the function will return `true` which would mean that the URL is forged which is not the case.

Reply:
```suggestion
  const whitelistedDapp = whitelistedDapps.find(({ dappId: currentDappId }) => currentDappId === dappId);

  if (!whitelistedDapp) {
    return false;
  }

  return !url.startsWith(whitelistedDapp.url);
```